### PR TITLE
SOHO-8076 - autocomplete `select()` return value bugfix

### PR DESCRIPTION
**Pull request by EdwardCoyle**
> Explain the **details** for making this change. What existing problem does the pull request solve?

a previous fix for SOHO-8076 was causing incorrect values to be returned as `selected` when a `data-index` attribute wasn't used in templates.  This has been fixed and a tests was added to demonstrate.

> **Related issue (required)**:

https://jira.infor.com/browse/SOHO-8076?focusedCommentId=4543384&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-4543384

> **Steps necessary to review your pull request (required)**:

- Pull the branch and run the functional tests for Autocomplete.  All should pass.

> Other Details:

Closes SOHO-8076

### CODE REVIEW

**Comment by tmcconechy** [comment]
Based on the fact that they have a patch manually. Please change this pr point to master.

See:
https://jira.infor.com/browse/SOHO-8076

File: test/components/autocomplete/autocomplete-select.func-spec.js [hunk 1]
```diff
@@ -64,7 +73,25 @@ describe('Autocomplete API (select)', () => {
     expect(selectedItem.value).toEqual('NJ'); // Should be retrieved from `data-value`
   });
 
-  it('returns an object with metadata about an item that was programmatically selected', () => {
```
Comment (d-g-h) [inline test/components/autocomplete/autocomplete-select.func-spec.js]: @tmcconechy here.

Comment (tmcconechy) [reply test/components/autocomplete/autocomplete-select.func-spec.js]: 👍 Ok will pull mine out

Comment (clepore) [reply test/components/autocomplete/autocomplete-select.func-spec.js]: Here?

Comment (clepore) [reply test/components/autocomplete/autocomplete-select.func-spec.js]: So I dunno what this convo was about, but @tmcconechy, when you approve this we can merge.

**Pull request merged**